src/Controls/tests/TestCases.HostApp/Issues/Issue29130.cs

```diff
@@ -0,0 +1,114 @@
+using System.Collections.ObjectModel;
+using Maui.Controls.Sample;
+
+namespace Maui.Controls.Sample.Issues;
+
+[Issue(IssueTracker.Github, 29130, "CollectionView2 ItemSizingStrategy should work for MeasureFirstItem", PlatformAffected.iOS)]
+public class Issue29130 : ContentPage
+{
+	CollectionView2 collectionView;
+	public Issue29130()
+	{
+		var viewModel = new Issue29130ViewModel();
+		this.BindingContext = viewModel;
+
+		var measureFirstItemButton = new Button
+		{
+			Text = "MeasureFirstItem",
+			AutomationId = "29130MeasureFirstItemButton"
+		};
+		measureFirstItemButton.Clicked += OnMeasureItemButtonClicked;
+
+		var measureAllItemsButton = new Button
+		{
+			Text = "MeasureAllItems",
+			AutomationId = "29130MeasureAllItemsButton"
+
+		};
+		measureAllItemsButton.Clicked += OnMeasureItemButtonClicked;
+
+		var grid = new Grid
+		{
+			AutomationId = "29130Grid",
+			ColumnDefinitions = new ColumnDefinitionCollection
+			{
+				new ColumnDefinition { Width = GridLength.Auto },
+				new ColumnDefinition { Width = GridLength.Auto }
+			}
+		};
+
+		// Add buttons to the grid with correct column assignments
+		Grid.SetColumn(measureFirstItemButton, 0);
+		Grid.SetColumn(measureAllItemsButton, 1);
```

Comment:
Could include another Button to change the size of the first item? For example, adding a `Height` property to the ItemModel or changing the `FontSize` property. This should invalidate and re-measure the cache.